posydon/popsyn/independent_sample.py

```diff
@@ -358,3 +358,49 @@ def generate_secondary_masses(primary_masses,
         secondary_masses = primary_masses
 
     return secondary_masses
+
+def binary_fraction_value(binary_fraction_const=1,binary_fraction_scheme = 'const',m1 = None,**kwargs):
+    """
+    Getting the binary fraction depending on the scheme. The two possible option are a constant binary fraction 
+    and a binary fraction based on the values given in Moe and Di Stefano (2017). 
+
+    Parameters:
+    --------------------
+    binary scheme: string
+        Determines if the value of the binary fraction will be constant or not  
+    binary fraction const: int 
+        Gives the value the constant value of the binary if the constant scheme is choosen. 
+
+    Returns 
+    ------------------
+    binary fraction: int
+
+    """
+    binary_fraction_scheme_options = ['const','Moe_17']
+
+    # Input parameter checks
+    if binary_fraction_scheme not in binary_fraction_scheme_options: 
+        raise ValueError("You must provide an allowed binary fraction scheme.")
+    
+
+    if binary_fraction_scheme == 'const': 
+        binary_fraction = binary_fraction_const
+    
+    elif binary_fraction_scheme == 'Moe_17':
+        if m1 is None: 
+            raise ValueError("There was not a primary mass provided in the inputs. Unable to return a binary fraction")
+        elif m1 <= 2  and m1 > 0.8:
+            binary_fraction = 0.4
+        elif m1 <= 5 and m1 > 2:
+            binary_fraction = 0.59
+        elif m1<=9 and m1 > 5 :
+            binary_fraction = 0.76
+        elif m1<= 16 and m1 > 9:
+            binary_fraction = 0.84
+        elif m1 > 16:
+            binary_fraction = 0.94
```

Comment:
Could you raise an Error for m1<0.8, currently this case is not covered at all.
May consider to convert the "pass" in case the scheme is not recognized into an Error raising, too.

Reply:
You are right, I will raise an error for this case.

Reply:
It is better to use an else clause instead of another elif, you now missed the m1=0.8 case.